posthog/views.py

```diff
@@ -80,11 +80,39 @@ def stats(request):
 
 
 def robots_txt(request):
-    ROBOTS_TXT_CONTENT = (
-        "User-agent: *\nDisallow: /shared_dashboard/\nDisallow: /shared/"
-        if is_cloud()
-        else "User-agent: *\nDisallow: /"
-    )
+    # Block all on self-hosted instances
+    if not is_cloud():
+        return HttpResponse("User-agent: *\nDisallow: /", content_type="text/plain")
+
+    ROBOTS_TXT_CONTENT = """User-agent: *
+
+# Block shared paths
+Disallow: /shared_dashboard/
+Disallow: /shared/
+
+# Block URLs with sensitive query parameters
+Disallow: /*?*email=
+Disallow: /*?*organization_name=
+Disallow: /*?*first_name=
+Disallow: /*?*token=
+Disallow: /*?*sharing_access_token=
+Disallow: /*%40*
+Disallow: /*@*
+
+# Block authentication paths
+Disallow: /login
+Disallow: /login/
+Disallow: /signup
+Disallow: /signup/
```

Comment:
This will imply Google wont learn about our login/signup pages, but that's a nope, I think people should be able to search for `posthog login` and find it

Reply:
Ok I can remove login/signup but keep the other pages.